api/data/seattle_db.go

```diff
@@ -2,53 +2,78 @@ package data
 
 import (
 	"context"
+	"fmt"
+	"time"
 
 	"github.com/gobuffalo/nulls"
 	"github.com/jackc/pgx/v4"
 )
 
 // SeattleOfficer is the object model for SPD officers
 type SeattleOfficer struct {
+	Date			time.Time    `json:"date,omitempty"`
 	BadgeNumber     string       `json:"badge_number,omitempty"`
-	FirstName       string       `json:"first_name,omitempty"`
-	MiddleName      nulls.String `json:"middle_name,omitempty"`
-	LastName        string       `json:"last_name,omitempty"`
+	FullName		string       `json:"full_name,omitempty"`
 	Title           string       `json:"title,omitempty"`
 	Unit            string       `json:"unit,omitempty"`
 	UnitDescription nulls.String `json:"unit_description,omitempty"`
+	FirstName       string       `json:"first_name,omitempty"`
+	MiddleName      nulls.String `json:"middle_name,omitempty"`
+	LastName        string       `json:"last_name,omitempty"`
 }
 
 // SeattleOfficerMetadata retrieves metadata describing the SeattleOfficer struct
-func (c *Client) SeattleOfficerMetadata() []map[string]string {
-	return []map[string]string{
-		{
-			"FieldName": "badge_number",
-			"Label":     "Badge Number",
-		},
-		{
-			"FieldName": "first_name",
-			"Label":     "First Name",
-		},
-		{
-			"FieldName": "middle_name",
-			"Label":     "Middle Name",
-		},
-		{
-			"FieldName": "last_name",
-			"Label":     "Last Name",
-		},
-		{
-			"FieldName": "title",
-			"Label":     "Title",
-		},
-		{
-			"FieldName": "unit",
-			"Label":     "Unit",
-		},
-		{
-			"FieldName": "unit_description",
-			"Label":     "Unit Description",
+func (c *Client) SeattleOfficerMetadata() DepartmentMetadata {
+	var date time.Time
+	err := c.pool.QueryRow(context.Background(),
+		`
+			SELECT date
+			FROM seattle_officers
+			WHERE date = (select max(date) from seattle_officers);
+		`).Scan(&date)
+
+	if err != nil {
+		fmt.Printf("DB Client Error: %s", err)
+		return DepartmentMetadata{}
+	}
+
+	return DepartmentMetadata{
+		Fields: []map[string]string{
+			{
+				"FieldName": "badge_number",
+				"Label":     "Badge Number",
+			},
+			{
+				"FieldName": "first_name",
+				"Label":     "First Name",
+			},
+			{
+				"FieldName": "middle_name",
+				"Label":     "Middle Name",
+			},
+			{
+				"FieldName": "last_name",
+				"Label":     "Last Name",
+			},
+			{
+				"FieldName": "title",
+				"Label":     "Title",
+			},
+			{
+				"FieldName": "unit",
+				"Label":     "Unit",
+			},
+			{
+				"FieldName": "unit_description",
+				"Label":     "Unit Description",
+			},
+			{
+				"FieldName": "full_name",
+				"Label":     "Full Name",
+			},
 		},
+		LastAvailableRosterDate: date.Format("2006-01-02"),
```

Comment:
Is this a typical date format call? Does the fact that this is 2006 matter? I'm a bit confused haha

Reply:
I wish there was a better way. I dunno why 2006 it was just in the code I copied and I didn't see a reason to change it. See below for more info:

https://golang.org/pkg/time/#pkg-constants

The constants section defines what formats are available and none of them were just YYYY-MM-DD so I went with that. Should probably set the day to be >12, though, to make it more obvious what's happening.

Reply:
Ok I tried changing it to `2006-01-02` and it started outputting like `2021-01-281` lol so apparently the 01-02 means "two digit month"-"two digit day".

Reply:
That is literally the dumbest thing I've ever seen, wow. Thanks for the context I guess haha